exercises/supervised-regression/ex_rnw/ex_regr_losses.Rnw

```diff
@@ -0,0 +1,75 @@
+In this exercise, we will examine loss functions for regression problems 
+somewhat more in-depth.
+
+<<echo=FALSE, message=FALSE, fig.align="left", fig.height=2.5, fig.width=4>>=
+
+set.seed(123L)
+x <- runif(100L)
+y <- 0.2 + 0.8 * x
+y <- y + rnorm(length(x), sd = 0.1)
+
+ggplot2::ggplot(data.frame(x = x, y = y), ggplot2::aes(x = x, y = y)) +
+  ggplot2::geom_point() + 
+  ggplot2::theme_bw() + 
+  ggplot2::geom_smooth(formula = y ~ x, method = "lm", se = FALSE) +
+  ggplot2::annotate("point", x = 0.95, y = 0.3, color = "orange", size = 2)
+
+@
+
+\begin{enumerate}[a)]
+  \item Consider the above linear regression problem. How will the model 
+  parameters be affected by adding the new outlier point (orange) if you use
+  \begin{enumerate}[i)]
+    \item $L1$ loss
+    \item $L2$ loss
+  \end{enumerate}
+  in your empirical risk? (You do not need to actually compute the 
+  parameter values.)
+\end{enumerate}
+
+<<echo=FALSE, message=FALSE, fig.align="left", fig.height=2.5, fig.width=4>>=
+
+huber_loss <- function(res, delta = 0.5) {
+  if (abs(res) <= delta) {
+    0.5 * (res^2)
+  } else {
+    delta * abs(res) - 0.5 * (delta^2)
+  }
+}
+
+x <- seq(-10L, 10L, length.out = 1000L)
+y <- sapply(x, huber_loss, delta = 5L)
+
+ggplot2::ggplot(data.frame(x = x, y = y), ggplot2::aes(x = x, y = y)) +
+  ggplot2::geom_line() + 
+  ggplot2::theme_bw()
+
+@
+
+\begin{itemize}
+  \item[b)] The second plot visualizes another loss function popular in 
+  regression tasks, the so-called \textit{Huber loss} (depending on 
+  $\delta > 0$; here: $\delta = 5$). 
+  Describe how the Huber loss deals with residuals as compared to $L1$ and $L2$ 
+  loss.
+  Can you guess its definition? 
+  \item[c)] \textcolor{orange}{Zu schwer?}
```

Comment:
Hm, ja ich weiß auch nicht. Da wir in der VL bisher auch nicht so mathematisch unterwegs sind, würde ich das hier glaube ich eher weglassen.